libraries/core_libs/consensus/include/final_chain/replay_protection_service.hpp

```diff
@@ -7,26 +7,32 @@
 
 namespace taraxa::final_chain {
 
-struct ReplayProtectionService {
+class ReplayProtectionService {
+ public:
   struct Config {
     uint64_t range = 0;
   };
 
+  ReplayProtectionService() = default;
   virtual ~ReplayProtectionService() = default;
+  ReplayProtectionService(const ReplayProtectionService &) = default;
+  ReplayProtectionService(ReplayProtectionService &&) = default;
+  ReplayProtectionService &operator=(const ReplayProtectionService &) = default;
+  ReplayProtectionService &operator=(ReplayProtectionService &&) = default;
 
-  virtual bool is_nonce_stale(const addr_t& addr, const trx_nonce_t& nonce) const = 0;
+  virtual bool is_nonce_stale(const addr_t &addr, const trx_nonce_t &nonce) const = 0;
 
   struct TransactionInfo {
     addr_t sender;
     trx_nonce_t nonce = 0;
   };
-  virtual void update(DB::Batch&, uint64_t, util::RangeView<TransactionInfo> const&) = 0;
+  virtual void update(DB::Batch &, uint64_t, util::RangeView<TransactionInfo> const &) = 0;
```

Comment:
if you already touching this you should change the order of const

Reply:
I am not, clang-format did format it automatically, which is weird that it did not format it before

Reply:
maybe you did update your clang suite?

Reply:
new clang-format new rules?

Reply:
no, something is wrong with clang-format as even in our existing code base, some function are formatted with & like here, e.g. https://github.com/Taraxa-project/taraxa-node/blob/develop/libraries/core_libs/consensus/include/dag/dag.hpp#L66

and some are formatted with & right after the type(without whitespace), e.g. https://github.com/Taraxa-project/taraxa-node/blob/develop/libraries/core_libs/consensus/include/dag/sortition_params_manager.hpp#L24

Reply:
I created [issue](https://github.com/Taraxa-project/taraxa-node/issues/1662) for it so this PR can be merged